lib/yocaml/metadata.ml

```diff
@@ -65,24 +73,19 @@ module Page = struct
     ]
   ;;
 
-  let from_yaml yaml =
-    let open Validate in
-    Yaml.as_object
-      yaml
-      (fun obj ->
-        let open Applicative in
-        make
-        <$> Yaml.(optional string "title" obj)
-        <*> Yaml.(optional string "description" obj))
-      (Validate.valid $ make None None)
-  ;;
-
-  let from_string = function
+  let from_string (module V : PROVIDER) = function
     | None -> Validate.valid $ make None None
     | Some str ->
-      Result.fold ~ok:from_yaml ~error:(function `Msg e ->
-          Error.(to_validate $ Yaml e))
-      $ Yaml.of_string str
+      let open Validate.Monad in
+      V.from_string str
+      >>= V.object_and (fun assoc ->
+              let open Validate.Applicative in
+              make
+              <$> V.(optional_assoc string) "title" assoc
+              <*> V.(optional_assoc string) "description" assoc)
+      |> (function
+      | Preface.Validation.Invalid _ -> Validate.valid $ make None None
+      | x -> x)
```

Comment:
Did you run the tests? I have the impression that the behaviour is slightly changed. Before this patch : 
``markdown
---
---
```
or
``markdown
---
I'm a title
---
```
was considered valid for page metadata. With the new feature, I think this is no longer the case.

Reply:
Yes, it is voluntary. To be honnest, I think it is better but I can switch to the old behaviour if you prefer.